coalib/processes/Processing.py

```diff
@@ -341,7 +345,15 @@ def instantiate_processes(section,
         log_printer,
         ignored_file_paths=glob_list(section.get('ignore', "")),
         limit_file_paths=glob_list(section.get('limit_files', "")))
+
+    # This stores all matched files irrespective of whether coala is run
+    # only on changed files or not. Global bears require all the files
+    complete_filename_list = filename_list
+    if section.get('caching', False):
+        filename_list = get_changed_files(filename_list, cache)
+
```

Comment:
you probably wanna make a comment here that the complete dict is given as the file dict while e.g. local bears are only run for every changed file. I.e. the whole project is accessible to every bear but tasks which are unchanged since last time don't change. This is pretty smart btw. good job.